docs/source/concepts/local-ledger.rst

```diff
@@ -0,0 +1,296 @@
+    
+    If an action `act`:sub:`u` uses a key that `A` contains a maintainer of
+    and there is a key allocation action `act`:sub:`a` ordered before `act`:sub:`u`,
+    then there is a deallocation action `act`:sub:`d` for the key ordered after `act`:sub:`a` and before `act`:sub:`u`.
+
+In the use ordering condition, the last activation before a contract use is well-defined
+because the graph is finite and all activations are totally ordered by the first condition.
+Moreover, a local ledger remains a local ledger if more edges between transaction vertices are added.
+
+The edges in a local ledger induce a partial order on the vertices, the **causality order**.
+A transaction `tx`:sub:`1` causally precedes `tx`:sub:`2` if there is a path from `tx`:sub:`1` to `tx`:sub:`2`.
+As the examples below will illustrate,
+the conditions ensure that the causality order captures at least the causal dependencies between transactions from `A`\ 's perspective.
+
+.. _order-consistency:
+
+Order consistency
+*****************
+
+The local ledger constrains the order of the transactions and transaction trees on the :ref:`Transaction Service <transaction-service>`.
+In detail, the following conditions hold:
+
+#. The transaction trees on `A`\ 's transaction tree stream are precisely those that are in the local ledger for `A`.
+#. The flat transaction stream contains precisely the ``CreatedEvent``\ s and ``ArchivedEvent``\ s
+   that correspond to **Create** and consuming **Exercise** actions in transaction trees on the transaction tree stream where `A` is a stakeholder of the affected contract.
+#. Whenever the local ledger causally orders two transactions, they appear in the same order on the streams.
+
+Similarly, the active contract service provides the set of contracts that are active at the returned offset according to the Transaction Service streams.
+That is, the effects of all events from the transaction event stream are taken into account in the provided set of contracts.
+In particular, an application can process all subsequent events from the flat transaction stream or the transaction tree stream without having to take events before the snapshot into account.
+
+A deployed DAML ledger ensures that all the local ledgers of its Participant Nodes are consistent, subject to the ledger's trust assumptions.
+Here, we consider only a single DAML ledger.
+The general case of :ref:`several interoperable DAML ledgers <interoperable-ledgers>` is discussed :ref:`elsewhere <interoperable-causality>`.
+Consistency can be expressed by embedding all the local ledgers into a single graph of transactions, the shared ledger.
+A shared ledger satisfies conditions similar to a local ledger, but the conditions do not restrict the orders to a party.
+
+Definition »shared ledgers«
+  A **shared ledger** is a finite directed acyclic graph that satisfies the following conditions:
```

Comment:
Yes, we could skip the `shared ledger` and directly go for the virtual shared ledger pretty much as you suggested.

I mainly included this because the reviewers on the previous draft criticized that I introduce a total order on the shared ledger: such a total order may create wrong expectations at the reader as some ledger implementations do not physically implement such a total order (e.g., Corda).